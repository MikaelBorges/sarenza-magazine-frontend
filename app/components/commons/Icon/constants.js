export const CHEVRON_RIGHT = 'CHEVRON_RIGHT';
export const CHEVRON_LEFT = 'CHEVRON_LEFT';
export const CHEVRON_DOWN = 'CHEVRON_DOWN';
export const SPINNER = 'SPINNER';
export const CHECKED = 'CHECKED';
export const CLOSE = 'CLOSE';
