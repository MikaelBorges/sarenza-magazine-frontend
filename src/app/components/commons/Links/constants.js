export const LINK_TYPE = {
  TEXT: 'test',
  BUTTON: 'button',
  OTHER: 'other'
};

export const ARROW_DIRECTION = {
  RIGHT: 'right',
  DOWN: 'down'
};
