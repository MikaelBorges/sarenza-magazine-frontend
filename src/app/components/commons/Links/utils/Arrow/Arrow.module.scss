.arrow svg {
  overflow: hidden;
  &path {
    fill: currentColor;
  }
}
