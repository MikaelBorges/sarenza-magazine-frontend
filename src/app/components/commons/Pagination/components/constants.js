export const TEXTS = {
  rowsPerPage: 'Lignes par page :',
  pageLabel: 'Page :',
  pageOn: 'sur',
  viewAll: 'Tout voir',
  next: 'Suiv',
  prev: 'Préc',
  nextButton: 'Suivant',
  prevButton: 'Précédent'
};

export const PAGE_RANGES = ['5', '10', '15', '20', '50', '100'];
