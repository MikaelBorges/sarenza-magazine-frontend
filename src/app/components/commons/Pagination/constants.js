export const TEXTS = {
  rowsPerPage: 'Lignes par page :',
  pageLabel: 'Page :',
  pageOn: 'sur',
  viewAll: 'Tout voir',
  next: 'Suiv',
  prev: 'Pr√©c'
};

export const PAGE_RANGES = ['5', '10', '15', '20', '50', '100'];
