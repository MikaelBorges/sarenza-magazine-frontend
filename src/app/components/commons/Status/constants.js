const STATUS_TYPE = {
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error'
};

export default STATUS_TYPE;
