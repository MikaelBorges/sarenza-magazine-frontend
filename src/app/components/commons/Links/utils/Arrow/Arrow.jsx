import PropTypes from 'prop-types';
import React from 'react';

import ChevronDown from '@/components/commons/Icon/svg/ChevronDown';
import ChevronRight from '@/components/commons/Icon/svg/ChevronRight';

import { ARROW_DIRECTION } from '../../constants';
import styles from './Arrow.module.scss';

const Arrow = ({ direction }) => {
  const arrowSize = '10px';

  switch (direction) {
    case ARROW_DIRECTION.DOWN:
      return (
        <>
          <span className={styles.arrow}>
            <ChevronDown width={arrowSize} height={arrowSize} />
          </span>
        </>
      );
    default:
      return (
        <>
          <span className={styles.arrow}>
            <ChevronRight width={arrowSize} height={arrowSize} />
          </span>
        </>
      );
  }
};

Arrow.propTypes = {
  direction: PropTypes.string
};

Arrow.defaultProps = {
  direction: ARROW_DIRECTION.RIGHT
};

export default Arrow;
