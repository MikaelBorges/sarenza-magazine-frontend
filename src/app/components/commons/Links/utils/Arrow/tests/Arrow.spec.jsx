import { render } from '@testing-library/react';
import React from 'react';

import { ARROW_DIRECTION } from '../../../constants';
import Arrow from '../Arrow';

describe('Arrow', () => {
  it('should match the expected snapshot with default props', () => {
    const wrapper = render(<Arrow />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match the expected snapshot with direction down', () => {
    const wrapper = render(<Arrow direction={ARROW_DIRECTION.DOWN} />);

    expect(wrapper).toMatchSnapshot();
  });
});
