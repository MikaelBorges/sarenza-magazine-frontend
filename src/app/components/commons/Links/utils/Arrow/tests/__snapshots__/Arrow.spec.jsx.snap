// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should match the expected snapshot with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="arrow"
      >
        <svg
          aria-labelledby="title desc"
          height="10px"
          role="img"
          viewBox="0 0 8.53 14.33"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title"
          />
          <desc
            id="desc"
          />
          <path
            d="M680.7 391.66l 7.16-7.09-7.16-7.23-1.36 1.36 5.8 5.87-5.8 5.73Z"
            fill="#fff"
            transform="translate(-679.34 -377.34)"
          />
        </svg>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="arrow"
    >
      <svg
        aria-labelledby="title desc"
        height="10px"
        role="img"
        viewBox="0 0 8.53 14.33"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title"
        />
        <desc
          id="desc"
        />
        <path
          d="M680.7 391.66l 7.16-7.09-7.16-7.23-1.36 1.36 5.8 5.87-5.8 5.73Z"
          fill="#fff"
          transform="translate(-679.34 -377.34)"
        />
      </svg>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Arrow should match the expected snapshot with direction down 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="arrow"
      >
        <svg
          aria-labelledby="title desc"
          height="10px"
          role="img"
          tabindex="0"
          viewBox="0 0 14 8"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title"
          />
          <desc
            id="desc"
          />
          <path
            d="M12.977,1.828 L7.670,7.135 C7.302,7.503 6.705,7.503 6.336,7.135 L1.030,1.828 C0.662,1.460 0.662,0.863 1.030,0.494 C1.398,0.126 1.995,0.126 2.364,0.494 L7.003,5.134 L11.643,0.494 C12.011,0.126 12.608,0.126 12.977,0.494 C13.345,0.863 13.345,1.460 12.977,1.828 Z"
            fill="#fff"
          />
        </svg>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="arrow"
    >
      <svg
        aria-labelledby="title desc"
        height="10px"
        role="img"
        tabindex="0"
        viewBox="0 0 14 8"
        width="10px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title"
        />
        <desc
          id="desc"
        />
        <path
          d="M12.977,1.828 L7.670,7.135 C7.302,7.503 6.705,7.503 6.336,7.135 L1.030,1.828 C0.662,1.460 0.662,0.863 1.030,0.494 C1.398,0.126 1.995,0.126 2.364,0.494 L7.003,5.134 L11.643,0.494 C12.011,0.126 12.608,0.126 12.977,0.494 C13.345,0.863 13.345,1.460 12.977,1.828 Z"
          fill="#fff"
        />
      </svg>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
